File: data/prefix_sum/prefix_sum.py
import random
import sys

# Fungsi untuk menghitung prefix sum
def prefix_sum(arr):
    prefix = [0] * len(arr)
    prefix[0] = arr[0]  # Elemen pertama

    # Menghitung prefix sum
    for i in range(1, len(arr)):
        prefix[i] = prefix[i - 1] + arr[i]

    return prefix

def main():
    # Mengecek apakah argumen ukuran array diberikan
    if len(sys.argv) != 2:
        print("Usage: python prefix_sum.py <size_of_array>")
        return

    try:
        # Mendapatkan ukuran array dari argumen input
        n = int(sys.argv[1])
    except ValueError:
        print("Ukuran array harus angka.")
        return

    if n <= 0:
        print("Ukuran array harus positif.")
        return

    # Menghasilkan array acak
    arr = [random.randint(0, 99) for _ in range(n)]
    print("Array acak:", arr)

    # Menghitung prefix sum
    result = prefix_sum(arr)
    print("Prefix Sum:", result)

if __name__ == "__main__":
    main()

File: data/prefix_sum/PrefixSum.java
import java.util.Random;

public class PrefixSum {

    // Fungsi untuk menghitung prefix sum
    public static void prefixSum(int[] arr, int[] prefix) {
        prefix[0] = arr[0]; // Elemen pertama

        // Menghitung prefix sum
        for (int i = 1; i < arr.length; i++) {
            prefix[i] = prefix[i - 1] + arr[i];
        }
    }

    public static void main(String[] args) {
        // Mengecek apakah argumen ukuran array diberikan
        if (args.length != 1) {
            System.out.println("Usage: java PrefixSum <size_of_array>");
            return;
        }

        // Mengambil ukuran array dari argumen eksekusi
        int n = Integer.parseInt(args[0]);

        // Validasi input
        if (n <= 0) {
            System.out.println("Ukuran array harus positif.");
            return;
        }

        // Membuat array acak
        int[] arr = new int[n];
        int[] prefix = new int[n];
        Random rand = new Random();

        // Mengisi array dengan bilangan acak dan menampilkannya
        System.out.println("Array acak:");
        for (int i = 0; i < n; i++) {
            arr[i] = rand.nextInt(100); // Bilangan acak antara 0 hingga 99
            System.out.print(arr[i] + " ");
        }
        System.out.println();

        // Menghitung prefix sum
        prefixSum(arr, prefix);

        // Menampilkan hasil prefix sum
        System.out.println("Prefix Sum:");
        for (int i = 0; i < n; i++) {
            System.out.print(prefix[i] + " ");
        }
        System.out.println();
    }
}
